src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -28,10 +44,143 @@ import {CDK_MENU, Menu} from './menu-interface';
     {provide: CDK_MENU, useExisting: CdkMenuBar},
   ],
 })
-export class CdkMenuBar extends CdkMenuGroup implements Menu {
+export class CdkMenuBar extends CdkMenuGroup implements Menu, AfterContentInit, OnDestroy {
   /**
    * Sets the aria-orientation attribute and determines where menus will be opened.
    * Does not affect styling/layout.
    */
   @Input('cdkMenuBarOrientation') orientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** All child MenuItem elements nested in this MenuBar. */
+  @ContentChildren(CdkMenuItem, {descendants: true})
+  private readonly _allItems: QueryList<CdkMenuItem>;
+
+  /** Event emitter used to end the subscription to document click events. */
+  private readonly _destroyDocumentSubscription: EventEmitter<void> = new EventEmitter();
+
+  /** A reference to the document. */
+  private readonly _document: Document;
+
+  /** Whether the element is inside of a ShadowRoot component. */
+  private _isInsideShadowRoot: boolean;
+
+  constructor(
+    private readonly _elementRef: ElementRef<HTMLElement>,
+    @Inject(DOCUMENT) document: any
+  ) {
+    super();
+
+    this._document = document;
+  }
+
+  ngAfterContentInit() {
+    super.ngAfterContentInit();
+
+    this._configChildSubscriptions();
+  }
+
+  /**
+   * Returns true if this MenuBar, or any open Menu triggered by a MenuItem in this MenuBar,
+   * contains the given HTMLElement.
+   * @param element the element to search for.
+   */
+  _contains(element: HTMLElement) {
+    return (
+      this._elementRef.nativeElement.contains(element) ||
+      this._allItems
+        .filter(menuItem => menuItem.hasMenu() && menuItem._menuTrigger!.isMenuOpen())
+        .some(menuItem => menuItem._menuTrigger!._menuPanel!._menu!._contains(element))
+    );
+  }
+
+  /**
+   * Configure child MenuItem subscriptions and ensure that future child elements are also
+   * subscribed to.
+   */
+  private _configChildSubscriptions() {
+    this._allItems
+      .filter(menuItem => menuItem.hasMenu())
+      .forEach(menuItem => this._subscribeToTriggers(menuItem));
```

Comment:
We've been advised against using forEach in our internal style guide. Does that apply to Angular code as well, @jelbourn?

Reply:
Only for arrays, this is a `QueryList`